### PR TITLE
Customize theme

### LINKED ISSUE #13: Create theme for dex-swap
Apply a new theme

Integrate front-end design into dex-uniswap (possibly incl. light mode)

### PR DESCRIPTION
**Pull request by biocom**
Closes #13 

Customize dex-swap theme

### CODE REVIEW

File: src/custom/components/Header/index.tsx [hunk 1]
```diff
@@ -1,8 +1,16 @@
...
 
 export const Header = styled(HeaderMod)`
   // CSS Override
+
+  ${UniIcon} {
+    display: flex;
+  }
+
+  ${HeaderFrame} {
+    border-bottom: 1px solid #3A3B5A;
```
Comment (biocom) [inline src/custom/components/Header/index.tsx:12]: @anxolin For some reason, this one doesn't get applied. Should I export HeaderFrame separately just like with Header?

e.g. `export const Header = styled(HeaderMod)`

Comment (anxolin) [reply src/custom/components/Header/index.tsx:12]: let me review it

Comment (anxolin) [reply src/custom/components/Header/index.tsx]: Here u have: 884d6e8a7b61cbafe69196b95907d4b13c61828c :)

**Comment by biocom** [comment]
General styles implemented, incl. buttons. For both Dark and Light mode.

Screen Shot 2020-11-02 at 16 09 15
Screen Shot 2020-11-02 at 16 09 08
Screen Shot 2020-11-02 at 16 09 00
Screen Shot 2020-11-02 at 16 08 48

**Comment by biocom** [comment]
Ready for review.

File: src/custom/components/Button/index.tsx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+  // CSS overrides
+  background-color: transparent;
+  background-image: ${({ theme }) => `linear-gradient(270deg, ${theme.purple} 30%, ${theme.blue1} 70%);`};
+  border: 0;
+
+  &:focus,
+  &:hover,
+  &:active {
+    background-color: transparent;
+    background-image: ${({ theme }) => `linear-gradient(270deg, ${theme.purple} 30%, ${theme.blue1} 70%);`};
```
Comment (biocom) [inline src/custom/components/Button/index.tsx:33]: Tried making the code dryer by putting this line into a variable. It's also used on line 26. Need help refactoring as the gradient will be likely re-used more and more. Perhaps it makes sense to define this variable in the general theme file instead.

Comment (anxolin) [reply src/custom/components/Button/index.tsx:33]: I see 2 options: 
- https://github.com/gnosis/dex-swap/pull/20

File: src/theme/index.tsx [hunk 3]
```diff
@@ -42,44 +42,47 @@ export function colors(darkMode: boolean): Colors {
 
     // text
     text1: darkMode ? '#FFFFFF' : '#000000',
-    text2: darkMode ? '#C3C5CB' : '#565A69',
+    text2: darkMode ? '#DCDCDC' : '#565A69',
```
Comment (anxolin) [inline src/theme/index.tsx:45]: We shouldn't modify any original file. I can help to make a "Mod" of this

Comment (biocom) [reply src/theme/index.tsx:45]: Yes, understand. There are multiple mods in this file just FYI.

File: src/theme/styled.d.ts [hunk 4]
```diff
@@ -44,6 +44,9 @@ export interface Colors {
   yellow1: Color
   yellow2: Color
   blue1: Color
+  purple: Color
+  border: Color
+  disabled: Color
```
Comment (Velenir) [inline src/theme/styled.d.ts:49]: As this is just a type we should be able to extend it fairly easily anywhere in `/custom`
[Merging Interfaces](https://www.typescriptlang.org/docs/handbook/declaration-merging.html#merging-interfaces)

File: src/theme/index.tsx (hunk 3 above)
Comment (Velenir) [reply src/theme/index.tsx:45]: I propose we create `custom/theme/index.tsx` with

```ts
import {colors as originColors} from '@/theme' // after #11 is merged

export function colors(darkMode: boolean): Colors {
  return {
    ...originColors(darkMode),
   <your_overrides>
}
```

**Review by anxolin — APPROVED** [review]
Good to go imo. @biocom can you just review the TODO from the background?

why is white?

**Review by Velenir — APPROVED** [review]

**Comment by anxolin** [comment]
@biocom I'm merging this one, cause it's already big and has a lot of approvals

if u can, check my comment, and we can address other things in new PRs https://github.com/gnosis/dex-swap/pull/8#pullrequestreview-522465697

**Pull request merged**